src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceOpProcessorImpl.scala

```diff
@@ -69,7 +75,8 @@ private[tracker] object InstanceOpProcessorImpl {
     private[this] def expungeInstance(id: Instance.Id)(implicit ec: ExecutionContext): Future[InstanceUpdateEffect] = {
       directInstanceTracker.instance(id).map {
         case Some(existingInstance: Instance) =>
-          InstanceUpdateEffect.Expunge(existingInstance)
+          val events = eventsGenerator.events(InstanceStatus.Killed, existingInstance, task = None, clock.now())
```

Comment:
A: Actually we can't. I didn't want to introduce a new state but obviously forgot the `TODO`. Background is that we need to publish an event here that notifies subscribers of _this instance is terminal_. When we forcefully expunge a `Running` task, `Running` is not sufficient for that. Thought about `Deleted`, but I need to further think about that. Thoughts on this?